### ISSUE TITLE
Git configuration in ort.yml

### ISSUE THREAD

**Issue by dgutson** [oss-review-toolkit/ort#6698]
We need to set some connection configuration of the vcs in general and git in particular, for example whether to use https instead of ssh.
Our current workaround is to place this in the docker entry point:
![IMG-20230317-WA0004](https://user-images.githubusercontent.com/9671770/225808992-f674fd2a-1a7b-4fdc-a37a-3dfbfc690672.jpg)

**Comment by sschuberth** [comment]
Similar feature requests occasionally came up in the past already, I remember @heliocastro having similar requirements. A configurable analyzer mechanism to do regex-based search & replace on URLs could also be used to replace hard-coded substitutions like not using `git://` anymore for GitHub repos, see

https://github.com/oss-review-toolkit/ort/blob/2e78d0f4f756b39afb61c8f1d542e5080849ef90/utils/ort/src/main/kotlin/Utils.kt#L180-L183

**Comment by fviernau** [comment]
> A configurable analyzer mechanism to do regex-based search & replace on URLs could also be used to replace hard-coded substitutions like not using `git://` anymore for GitHub repos, see

Not that this needs to work for submodules as well.

**Comment by dgutson** [comment]
@fviernau s/not/note/ ?

If so, indeed I think a substitution wouldn't work for automatic submodules cloning, where ORT does not control the URL.

**Comment by dgutson** [comment]
I found more configuration rules that we had to set, which I'm not sure that can be specified as regexes:

```
git config --global url."https://github.com/".insteadOf ssh://git@github.com/
git config --global url."https://github.com/".insteadOf git@github.com:
```

**Comment by sschuberth** [comment]
Thanks to https://github.com/oss-review-toolkit/ort/pull/6387 this could probably quite elegantly implemented as a special curation providers that does regex search & replaces on URLs from package metadata.

**Comment by sschuberth** [comment]
> Note that this needs to work for submodules as well.

True, so any such URL replacement would also need to be applied by the downloader, not only the analyzer. A repo to test this with could be https://github.com/TritonDataCenter/node-http-signature/blob/v1.2.0/.gitmodules as it uses SSH / SCP URLs like `url = git@github.com:davepacheco/javascriptlint` for submodules.

**Comment by sschuberth** [comment]
URL replacements for the *downloader* (that also work for submodule URLs in a dependency's Git repository) have been fixed / fully implemented in https://github.com/oss-review-toolkit/ort/pull/11094. However, this is not configurable yet for end-users of the CLI yet, which is tracked by https://github.com/oss-review-toolkit/ort/issues/8556.

Regarding URL replacements for the *analyzer*, by [now](https://github.com/oss-review-toolkit/ort/wiki/ORT-Weekly-Meeting#2025-11-20), the recommendation is to create a custom [PackageCurationProvider plugin](https://github.com/oss-review-toolkit/ort/blob/main/plugins/package-curation-providers/api/src/main/kotlin/PackageCurationProvider.kt) that replaces URLs dynamically to your liking.

I'm keeping this open until the new `replaceSshWithHttps` option can be set via ORT's `config.yml`.
